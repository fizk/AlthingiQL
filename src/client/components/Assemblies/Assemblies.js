import React from 'react';
import PropTypes from 'prop-types';
import {AssemblyBadge} from '../../elements/AssemblyBadge';
import {ListSeparated, ListItem} from '../../elements/List';

export default class Index extends React.Component {
    static propTypes = {
        assemblies: PropTypes.arrayOf(PropTypes.shape({
            id: PropTypes.number,
            period: PropTypes.shape({
                from: PropTypes.string,
                to: PropTypes.string,
            }),
            division: PropTypes.shape({
                majority: PropTypes.arrayOf(PropTypes.shape({
                    id: PropTypes.number,
                    name: PropTypes.string,
                    color: PropTypes.string,
                })),
                minority: PropTypes.arrayOf(PropTypes.shape({
                    id: PropTypes.number,
                    name: PropTypes.string,
                    color: PropTypes.string,
                }))
            })
        }))
    };

    static defaultProps = {
        assemblies: []
    };

    render() {
        return (
            <ListSeparated>
                {this.props.assemblies.map(assembly => (
                    <ListItem key={`assembly-${assembly.id}`} >
                        <AssemblyBadge assembly={assembly} />
                    </ListItem>
                ))}
            </ListSeparated>
        )
    }
}
