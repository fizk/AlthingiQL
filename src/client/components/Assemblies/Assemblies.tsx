import * as React from "react";
import AssemblyBadge from "../../elements/AssemblyBadge";
import { ListSeparated, ListItem } from "../../elements/List";

type IndexProps = {
    assemblies: {
        id: number,
        period: {
            from?: string,
            to?: string
        },
        division: {
            majority?: {
                id?: number,
                name?: string,
                color?: string
            }[],
            minority: {
                id?: number,
                name?: string,
                color?: string
            }[]
        }
    }[]
    loading?: boolean
};

export default class Index extends React.Component<IndexProps, {}> {
    static defaultProps: IndexProps = {
        assemblies: [],
        loading: false
    };

    render() {
        return (
            <ListSeparated>
                {this.props.assemblies.map(assembly => (
                    <ListItem key={`assembly-${assembly.id}`}>
                        <AssemblyBadge assembly={assembly} />
                    </ListItem>
                ))}
            </ListSeparated>
        );
    }
}
