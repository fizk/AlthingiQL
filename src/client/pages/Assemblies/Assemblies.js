import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom'

export default class Index extends React.Component {
    static propTypes = {
        assemblies: PropTypes.arrayOf(PropTypes.shape({
            id: PropTypes.number,
            period: PropTypes.shape({
                from: PropTypes.string,
                to: PropTypes.string,
            })
        }))
    };

    static defaultProps = {
        assemblies: []
    };

    render() {
        return (
            <ul>
                {this.props.assemblies.map(assembly => (
                    <li key={`assembly-${assembly.id}`}>
                        <Link to={`/loggjafathing/${assembly.id}`}>{assembly.id}</Link>
                        {assembly.period.from} {assembly.period.to}
                    </li>
                ))}
            </ul>
        )
    }
}
