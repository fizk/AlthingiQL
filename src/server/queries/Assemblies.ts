import Assembly from '../types/Assembly';
import {GraphQLList} from 'graphql';

export default {
    type: new GraphQLList(Assembly),

    resolve(root, params, {client}) {
        return client.get('/loggjafarthing');
    },
};
