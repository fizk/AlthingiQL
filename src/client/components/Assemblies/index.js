import {graphql, compose, gql} from 'react-apollo';
import Assemblies from './Assemblies';

const assembliesQuery = gql`
    query {
        Assemblies {
            id
            period {from to}
            division {
                majority {id name color}
                minority {id name color}
            }
        }
    }
`;

export default compose(
    graphql(assembliesQuery, {
        props: all => ({
            assemblies: all.data.loading === false ? all.data.Assemblies : undefined,
            loading: all.data.loading,
        })
    })
)(Assemblies);

