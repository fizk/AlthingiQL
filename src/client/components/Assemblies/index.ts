import {graphql, compose, gql} from 'react-apollo';
import Assemblies from './Assemblies';
import {AssemblyStatistics} from '../../../../@types';

const assembliesQuery = gql`
    query {
        Assemblies {
            id
            period {from to}
            division {
                majority {id name color}
                minority {id name color}
            }
        }
    }
`;

export default compose<any>( //@todo `any`
    graphql(assembliesQuery, {
        props: (all: {data?: {loading: boolean, Assemblies: AssemblyStatistics}}) => ({
            assemblies: all.data.loading === false ? all.data.Assemblies : undefined,
            loading: all.data.loading,
        }),
    }),
)(Assemblies);
