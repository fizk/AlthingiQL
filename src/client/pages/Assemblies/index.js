import {graphql, compose, gql} from 'react-apollo';
import Assemblies from './Assemblies';

const assembliesQuery = gql`
    query {
        Assemblies {
            id
            period {
                from
                to
            }
        }
    }
`;

export default compose(
    graphql(assembliesQuery, {
        props: all => ({
            assemblies: all.data.loading === false ? all.data.Assemblies : undefined,
            loading: all.data.loading,
        })
    })
)(Assemblies);

