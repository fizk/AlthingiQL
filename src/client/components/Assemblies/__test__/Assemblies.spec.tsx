import * as React from 'react';
import Assemblies from '../Assemblies';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

describe('<Assemblies />', () => {
    test('default props', () => {
        const tree = renderer.create(
            <Router >
                <Assemblies assemblies={undefined} />
            </Router>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('data', () => {
        const assemblies = [{
            id: 1,
            period: {
                from: '2001-01-01',
                to: '2001-01-01',
            },
            division: {
                majority: [{
                    id: 2,
                    name: 'party',
                    color: '123123',
                }],
                minority: [{
                    id: 2,
                    name: 'party',
                    color: '123123',
                }]
            }
        }];
        const tree = renderer.create(
            <Router >
                <Assemblies assemblies={assemblies} />
            </Router>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});


