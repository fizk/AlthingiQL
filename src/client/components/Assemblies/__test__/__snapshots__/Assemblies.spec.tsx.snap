// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Assemblies /> data 1`] = `
<ul
  className="list--separated"
>
  <li
    className="list__item"
  >
    <a
      aria-current={null}
      className="assembly-badge"
      href="/loggjafarthing/1"
      onClick={[Function]}
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <h3>
          1
        </h3>
        <time>
          2001
           
          -
           
          2001
        </time>
      </div>
      <ul
        className="list--horizontal"
      >
        <li
          className="list__item"
        >
          <div
            className="badge badge--sm"
            style={
              Object {
                "backgroundColor": "#123123",
              }
            }
            title="party"
          />
        </li>
      </ul>
    </a>
  </li>
</ul>
`;

exports[`<Assemblies /> default props 1`] = `
<ul
  className="list--separated"
/>
`;
